rtengine/impulse_denoise.h

```diff
@@ -42,14 +44,14 @@ SSEFUNCTION void ImProcFunctions::impulse_nr (LabImage* lab, double thresh)
 
     // buffer for the lowpass image
     float ** lpf = new float *[height];
+    lpf[0] = new float [width * height];
     // buffer for the highpass image
-    float ** impish = new float *[height];
+    char ** impish = new char *[height];
+    impish[0] = new char [width * height];
 
-    for (int i = 0; i < height; i++) {
-        lpf[i] = new float [width];
-        //memset (lpf[i], 0, width*sizeof(float));
-        impish[i] = new float [width];
-        //memset (impish[i], 0, width*sizeof(unsigned short));
+    for (int i = 1; i < height; i++) {
```

Comment:
Nice trick to fix the array-of-arrays problem without changing the users. Maybe this merits a separate helper class that does this transformation and manages the memory using constructors and destructors?